daemon/src/model/cfd.rs

```diff
@@ -1276,6 +1279,273 @@ impl Cfd {
+    pub struct CfdAggregate {
+        version: u64,
+
+        // static
+        uuid: Uuid,
+        position: Position,
+        initial_price: Price,
+        leverage: Leverage,
+        settlement_time_interval_hours: Duration,
+        quantity: Usd,
+
+        // TODO: plug in once we model this
+        // pub counterparty_network_identity: Uuid,
+
+        // dynamic (based on events)
+        dlc: Option<Dlc>,
+        cet: Option<Cet>,
+        attestation: Option<Attestation>,
+
+        revoke_commit: Option<RevokedCommit>,
+
+        // TODO: Proabbly better to bundle this in some structure?
+        lock_finality: bool,
+        commit_finality: bool,
+        refund_finality: bool,
+        cet_finality: bool,
+        collaborative_settlement_finality: bool,
+
+        cet_timelock_expired: bool,
+        refund_timelock_expired: bool,
+
+        // TODO: Can we map to cfd state for now to have contained changes in the legacy code?
+        cfd_state: CfdState,
+    }
+
+    impl CfdAggregate {
+        pub fn new(
+            uuid: Uuid,
+            position: Position,
+            initial_price: Price,
+            leverage: Leverage,
+            settlement_time_interval_hours: Duration,
+            quantity: Usd,
+        ) -> Self {
+            CfdAggregate {
+                version: 1,
+                uuid,
+                position,
+                initial_price,
+                leverage,
+                settlement_time_interval_hours,
+                quantity,
+                dlc: None,
+                cet: None,
+                attestation: None,
+                revoke_commit: None,
+                lock_finality: false,
+                commit_finality: false,
+                refund_finality: false,
+                cet_finality: false,
+                collaborative_settlement_finality: false,
+                cet_timelock_expired: false,
+                refund_timelock_expired: false,
+                cfd_state: CfdState::outgoing_order_request(),
+            }
+        }
+
+        pub fn start_contract_setup(&self) {
+            todo!(
+                "should this advance state? if it does not then there is not point in having this"
+            )
+        }
+
+        // TODO: Opted for one single function that handles all commands rather than a lot of
+        // different functions.  This makes it easier to see all commands and have the logic
+        // in one place. But it is one big function again - pretty cleaned up though.
+        //  Somewhat less typing when implementing it like this, not sure if better than having
+        // separate functions...  Separate functions might be more readable...?
+        pub fn handle_command(&self, command: Command) -> Vec<Event> {
+            match command {
+                Command::CompleteContractSetup(result) => match result {
+                    Ok(dlc) => vec![Event::ContractSetupCompleted(dlc)],
```

Comment:
These commands will need to validate that executing them on the current state makes sense.

Reply:
Should the `handle_command` function be able to fail? Otherwise, in a case where I don't get an event (but expect one...?) - what should the behavior be?